sdk/messaging/azservicebus/message.go

```diff
@@ -15,42 +15,109 @@ import (
 
 // ReceivedMessage is a received message from a Client.NewReceiver().
 type ReceivedMessage struct {
+	// MessageID is an application-defined value that uniquely identifies
+	// the message and its payload. The identifier is a free-form string.
+	//
+	// If enabled, the duplicate detection feature identifies and removes further submissions
+	// of messages with the same MessageId.
 	MessageID string
 
-	ContentType      *string
-	CorrelationID    *string
-	SessionID        *string
-	Subject          *string
-	ReplyTo          *string
+	// ContentType describes the payload of the message, with a descriptor following
+	// the format of Content-Type, specified by RFC2045 (ex: "application/json").
+	ContentType *string
+
+	// CorrelationID allows an application to specify a context for the message for the purposes of
+	// correlation, for example reflecting the MessageID of a message that is being
+	// replied to.
+	CorrelationID *string
+
+	// SessionID is used with session-aware entities and associates a message with an application-defined
+	// session ID. Note that an empty string is a valid session identifier.
+	// Messages with the same session identifier are subject to summary locking and enable
+	// exact in-order processing and demultiplexing. For session-unaware entities, this value is ignored.
+	SessionID *string
+
+	// Subject enables an application to indicate the purpose of the message, similar to an email subject line.
+	Subject *string
+
+	// ReplyTo is an application-defined value specify a reply path to the receiver of the message. When
+	// a sender expects a reply, it sets the value to the absolute or relative path of the queue or topic
+	// it expects the reply to be sent to.
+	ReplyTo *string
+
+	// ReplyToSessionID augments the ReplyTo information and specifies which SessionId should
+	// be set for the reply when sent to the reply entity.
 	ReplyToSessionID *string
-	To               *string
 
+	// To is reserved for future use in routing scenarios but is not currently used by Service Bus.
+	// Applications can use this value to indicate the logical destination of the message.
+	To *string
```

Comment:
This is basically what everyone has written - the intended meaning is that this field is "reserved for future use" but isn't currently being used by Service Bus. So you can use it, just with that caveat.

We can't omit the field since there's nothing preventing customers from using it and all the other SDKs make it available, so we'd basically be preventing them from getting that value at all in our SDK, which is bad for interop.

Reply:
Okay, talked with @chlowell. We'll leave it as is for now.